internal/saptune/saptune_test.go

```diff
@@ -0,0 +1,232 @@
+
+	suite.Error(err)
+	suite.ErrorContains(err, "saptune version not supported")
+}
+
+func (suite *SaptuneClientTestSuite) TestSuccessfulSaptuneVersionCheck() {
+	ctx := context.Background()
+
+	suite.mockExecutor.On(
+		"Exec",
+		saptuneVersionRetrieverCommandArguments(ctx)...,
+	).Return([]byte("3.1.0"), nil)
+
+	saptuneClient := saptune.NewSaptuneClient(
+		suite.mockExecutor,
+		suite.logger,
+	)
+	err := saptuneClient.CheckVersionSupport(ctx)
+
+	suite.NoError(err)
+}
+
+func (suite *SaptuneClientTestSuite) TestGettingAppliedSolutionFailure() {
+	ctx := context.Background()
+
+	suite.mockExecutor.On(
+		"Exec",
+		saptuneAppliedSolutionCommandArguments(ctx)...,
+	).Return(nil, errors.New("error calling saptune"))
+
+	saptuneClient := saptune.NewSaptuneClient(
+		suite.mockExecutor,
+		suite.logger,
+	)
+	appliedSolution, err := saptuneClient.GetAppliedSolution(ctx)
+
+	suite.Error(err)
+	suite.ErrorContains(err, "could not call saptune solution applied: error calling saptune")
+	suite.Empty(appliedSolution)
+}
+
+func (suite *SaptuneClientTestSuite) TestGettingNoSolutionApplied() {
+	ctx := context.Background()
+
+	noSolutionApplied := helpers.ReadFixture("saptune/applied_no_solution.json")
+
+	suite.mockExecutor.On(
+		"Exec",
+		saptuneAppliedSolutionCommandArguments(ctx)...,
+	).Return(noSolutionApplied, nil)
+
+	saptuneClient := saptune.NewSaptuneClient(
+		suite.mockExecutor,
+		suite.logger,
+	)
+	appliedSolution, err := saptuneClient.GetAppliedSolution(ctx)
+
+	suite.NoError(err)
+	suite.Empty(appliedSolution)
+}
+
+func (suite *SaptuneClientTestSuite) TestGettingAppliedSolution() {
+	ctx := context.Background()
+
+	hanaSolutionApplied := helpers.ReadFixture("saptune/applied_hana_solution.json")
+
+	suite.mockExecutor.On(
+		"Exec",
+		saptuneAppliedSolutionCommandArguments(ctx)...,
+	).Return(hanaSolutionApplied, nil)
+
+	saptuneClient := saptune.NewSaptuneClient(
+		suite.mockExecutor,
+		suite.logger,
+	)
+	appliedSolution, err := saptuneClient.GetAppliedSolution(ctx)
+
+	suite.NoError(err)
+	suite.Equal("HANA", appliedSolution)
+}
+
+func (suite *SaptuneClientTestSuite) TestApplySolutionFailureBecauseCommandFails() {
+	ctx := context.Background()
+
+	suite.mockExecutor.On(
+		"Exec",
+		saptuneApplySolutionCommandArguments(ctx, "HANA")...,
+	).Return(nil, errors.New("error calling saptune"))
+
+	saptuneClient := saptune.NewSaptuneClient(
+		suite.mockExecutor,
+		suite.logger,
+	)
+	err := saptuneClient.ApplySolution(ctx, "HANA")
+
+	suite.Error(err)
+	suite.ErrorContains(err, `could not perform saptune apply solution HANA`)
+}
+
+func (suite *SaptuneClientTestSuite) TestApplySolutionFailureBecauseAnAlreadyAppliedSolution() {
+	ctx := context.Background()
+
+	alreadyAppliedSolution := helpers.ReadFixtureString("saptune/apply_already_applied_solution.output")
+
+	suite.mockExecutor.On(
+		"Exec",
+		saptuneApplySolutionCommandArguments(ctx, "HANA")...,
+	).Return(nil, errors.New(alreadyAppliedSolution))
```

Comment:
Is this correct?
I mean, the `already_applied_solution.output`, should be the first value to return, because it is the output. And the second could be something like `exit status: 1`, or whatever.

Reply:
Fixed, thanks!